rescan_datastores: refactoring

rkhozinov opened the pull request.

Commit:
rescan_datastores: refactoring

ekhomyakova reviewed (changes requested):
I don't like it

Commit:
rescan: refactoring

ekhomyakova commented on servers/partner-ci/builders/rescan_ds.py:
```diff
@@ -47,360 +41,83 @@ class Victl(object):
-    if inst.check_portgroup_configured(args.datacenter, args.cluster,
-                                       args.portgroup):
+        self.dc_name = dc_name
+        self.datacenter = dc_name
```
What is the difference between `self.dc_name` and `self.datacenter`?

ekhomyakova commented on servers/partner-ci/builders/rescan_ds.py:
```diff
@@ -47,360 +41,83 @@ class Victl(object):
+
+    def rescan_datastores(self):
+        """Rescan all datastores."""
+        [host.configManager.storageSystem.RescanAllHba() for host in self.hosts]
```
Don't create new object. There is nothing bad in usual loop.

ekhomyakova commented on servers/partner-ci/builders/rescan_ds.py:
```diff
@@ -47,360 +41,83 @@ class Victl(object):
+            host_view = self.content.viewManager.CreateContainerView(self.datacenter,
+                                                                     [vim.HostSystem],
+                                                                     True)
+            self._hosts = [host for host in host_view.view ]
```
Redundant space

ekhomyakova commented on servers/partner-ci/builders/rescan_ds.py:
```diff
@@ -47,360 +41,83 @@ class Victl(object):
+
+    @property
+    def hosts(self):
+
```
Remove the line

ekhomyakova commented on servers/partner-ci/builders/rescan_ds.py:
```diff
@@ -47,360 +41,83 @@ class Victl(object):
+    def hosts(self):
+
+        if not self._hosts:
+            host_view = self.content.viewManager.CreateContainerView(self.datacenter,
```
Move `self.datacenter,` to the new line.

ekhomyakova commented on servers/partner-ci/builders/rescan_ds.py:
```diff
@@ -47,360 +41,83 @@ class Victl(object):
 if __name__ == '__main__':
-    args = Params()
-    inst = Victl(args.host, args.user, args.password, args.port)
 
```
Remove the line

rkhozinov replied:
done

ekhomyakova commented on servers/partner-ci/builders/rescan_ds.py:
```diff
@@ -47,360 +41,83 @@ class Victl(object):
-        args.cluster = cluster
-        datastore_list(args, inst)
+    victl = Victl(host=host, dc_name=dc_name, user=user,
+                    password=password, port=port)
```
Edit formatting

ekhomyakova replied:
Add empty line after last one.

Commit:
travis: flake8 checker is added for python code

Commit:
rescan_ds: add a fix to run with virtualenv libs

rkhozinov committed:
Merge pull request #154 from rkhozinov/travis-flake

travis: flake8 checker is added for python code

rkhozinov commented:
recheck

rkhozinov committed:
Update rescan_ds.py

Commit:
Merge branch 'master' into rescan-refactoring

Conflicts:
	.travis.yml

Commit:
rescan_ds: style fixes related to flake8

naumvd95 committed:
fix to avoid venv-activating during python-build-step

proofs
http://jenkins-tpi.bud.mirantis.net:8080/view/9.x.dvs/job/9.x.dvs.custom.qa/100/console

naumvd95 committed:
Merge branch 'rescan-refactoring' of github.com:rkhozinov/jenkins-jobs into rescan-refactoring

rkhozinov committed:
travis: fix checking of undefined jjb variables

rkhozinov commented on servers/partner-ci/builders/rescan_ds.py:
```diff
@@ -21,7 +21,8 @@
 
-sys.path.append(path.join(sys.path[0], "{0}/bin".format(environ.get('VIRTUAL_ENV'))))
+jenkins_venv = "/home/jenkins/90-venv/bin/activate_this.py"
+execfile(jenkins_venv, dict(__file__=jenkins_venv))
```
awesome

rkhozinov commented on .travis.yml:
```diff
@@ -8,18 +8,20 @@ python:
   - bash -c 'shopt -s globstar; shellcheck -e SC1091,SC2086,SC2046,SC2015,SC2155,SC1090,SC2012,SC2035,SC2016 **/*.sh'
   - tox -e 9.x,9.1,9.0
-  - bash -c 'grep -m1 -n -o --color -R -E "[^$]{(\w\w+)}" ./output && exit 1 || echo "there is no undefined variables"'
+  - bash -c 'grep --color -R -E "[^$]{[a-zA-Z0-9_-]+}" ./output && exit 1 || echo "there is no undefined variables"'
```
not good 
that merge has broken travis

ekhomyakova commented on .travis.yml:
```diff
@@ -8,18 +8,20 @@ python:
   apt:
     sources:
-      - debian-sid    # Grab shellcheck from the Debian repo
+      - debian-sid    # Grab shellcheck from the Debian repo (o_O)
```
Remove " (o_O)"

ekhomyakova commented on servers/partner-ci/builders/rescan_ds.py:
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env python
+
```
Is it needed?

rkhozinov replied:
@naumvd95 please remove that

ekhomyakova commented on servers/partner-ci/builders/rescan_ds.py:
```diff
@@ -14,26 +15,24 @@
+from os import path
 
-import paramiko
+jenkins_venv = "/home/jenkins/90-venv/bin/activate_this.py"
```
Don't hardcode path

rkhozinov replied:
@naumvd95 remove hardcode for virtual_env
use VIRTUAL_ENV variable instead

ekhomyakova commented on servers/partner-ci/builders/rescan_ds.py:
```diff
@@ -47,360 +46,74 @@ class Victl(object):
+    @property
+    def hosts(self):
+        if not self._hosts:
+            host_view = self.content.viewManager.CreateContainerView(self.datacenter,
```
Move `self.datacenter,` to the next line

rkhozinov replied:
flake check has passed with openstack guidelines

ekhomyakova commented on servers/partner-ci/builders/rescan_ds.py:
```diff
@@ -47,360 +46,74 @@ class Victl(object):
-                                     args.datastore):
+    def rescan_datastores(self):
+        """Rescan all datastores."""
+        [host.configManager.storageSystem.RescanAllHba() for host in self.hosts]
```
Make it in a loop. Generators are not suitable in this case

rkhozinov replied:
Please link a proof

naumvd95 committed:
refactoring venv-system, restructure rescan nfs case

1)direct injecting venv_path as a common var for all jobs
2)activating venv from python step
3) bug-fixes
proofs
http://jenkins-tpi.bud.mirantis.net:8080/view/9x-custom/job/9.x.vcenter.custom.qa/94/console

naumvd95 committed:
Merge branch 'rescan-refactoring' of github.com:rkhozinov/jenkins-jobs into rescan-refactoring

rkhozinov commented on servers/partner-ci/builders/common.prepare.sh:
```diff
@@ -52,7 +52,7 @@ fi
 
 export ENV_NAME="${ENV_PREFIX}.${SNAPSHOTS_ID}"
-export VENV_PATH="${HOME}/${FUEL_RELEASE}-venv"
+[ -z ${VENV_PATH} ] && { export VENV_PATH="${HOME}/${FUEL_RELEASE}-venv"; }
```
`export VENV_PATH=${VENV_PATH:-"$HOME/$FUEL_RELEASE-venv"}`

rkhozinov commented on servers/partner-ci/builders/common.prepare.sh:
```diff
@@ -251,3 +251,6 @@ fi
 [[ "${DEBUG}" == "true" ]] && virsh list --all
 sudo cp /var/log/libvirt/libvirtd.log ${WORKSPACE}/libvirtd_before_test.log
 sudo chown jenkins:jenkins ${WORKSPACE}/libvirtd_before_test.log
+
```
remove empty lines

rkhozinov commented on servers/partner-ci/builders/rescan_ds.py:
```diff
@@ -14,26 +15,23 @@
-
-import paramiko
-
+from os import path
```
add an empty line

rkhozinov commented on servers/partner-ci/common.yaml:
```diff
@@ -128,6 +128,7 @@
             SNAPSHOT_OUTPUT_FILE=extra_repos.sh;
             WORKSTATION_PASSWORD=VMware01;
             WORKSTATION_USERNAME=vmware;
+            VENV_PATH=/home/jenkins/90-venv;
```
try to use `$HOME` instead

naumvd95 committed:
refactor-cases , as @rkhozinov decision

ekhomyakova commented on servers/partner-ci/builders/rescan_ds.py:
```diff
@@ -14,26 +14,25 @@
 import sys
-
 from os import environ
+from os import path
```
Seems you don't use it.

ekhomyakova commented on servers/partner-ci/builders/rescan_ds.py:
```diff
@@ -47,360 +46,73 @@ class Victl(object):
     content = None
+    _datacenter = None
+    _hosts = []
+    _clusters = []
```
Seems you don't use it

naumvd95 committed:
code-style fix to avoid travis checks

Commit:
Refactoring according with pep8

Commit:
Fix venv activating

rkhozinov reviewed (commented):
looks good for me
